test/operators.jl

```diff
@@ -58,6 +58,25 @@ p = 1=>:foo
 @test_throws ArgumentError Base.scalarmax('a',['c','d'])
 @test_throws ArgumentError Base.scalarmax(['a','b'],'c')
 
+s1 = Set([1])
+s2 = Set([2])
+@test_throws MethodError min(s1, s2)
+@test_throws MethodError max(s1, s2)
+@test_throws MethodError minmax(s1, s2)
+
+struct TO
+  x
+end
+Base.isless(a::TO, b::TO) = isless(a.x, b.x)
+Base.isequal(a::TO, b::TO) = isequal(a.x, b.x)
+
+@test min(TO(1), TO(2)) == TO(1)
```

Comment:
What I meant is that that is already working on master:
```julia
julia> using Base.Test

julia> struct TO
         x
       end

julia> Base.isless(a::TO, b::TO) = isless(a.x, b.x)

julia> Base.isequal(a::TO, b::TO) = isequal(a.x, b.x)

julia> @testset begin
           @test min(TO(1), TO(2)) == TO(1)
           @test min(TO(2), TO(1)) == TO(1)
           @test max(TO(1), TO(2)) == TO(2)
           @test max(TO(2), TO(1)) == TO(2)
           @test minmax(TO(1), TO(2)) == (TO(1), TO(2))
           @test minmax(TO(2), TO(1)) == (TO(1), TO(2))
       end
Test Summary: | Pass  Total
test set      |    6      6
```
Becase `<` for `TO` falls back to https://github.com/JuliaLang/julia/blob/ed746ed14100c564884e525b9f830ed5cdd01b3f/base/operators.jl#L200

It is fine to test working code, but I guess the point was to test something that this PR fixes? In that case the test case should be updated :)